### PR TITLE
Deprecate Original JS Interop Helpers CPLAT-4909

### PR DESCRIPTION
**Pull request by joebingham-wk**
## Description:
Add deprecation messages for methods within `js_interop_helpers.dart` in favor of using `js_util` and add `jsifyAndAllowInterop`.

## Changes
- Deprecate `jsify`, `EmptyObject`, `setProperty`, and `getProperty`
- Add `jsifyAndAllowInterop` and  `_convertDataTree`
- Add test coverage for `jsifyAndAllowInterop`.

## Testing suggestions:
- CI Passes

## Potential areas of regression:
None, changes were simply deprecation methods and new methods.

### CODE REVIEW

File: lib/react_client/js_interop_helpers.dart [hunk 1]
```diff
@@ -102,3 +116,85 @@ EmptyObject jsify(Map map) {
...
+/// methods in this library. Only use this method as a last resort.
+///
+/// Recursively converts a JSON-like collection of Dart objects to a
+/// collection of JavaScript objects and returns a [JsObject] proxy to it.
+///
+/// [object] must be a [Map] or [Iterable], the contents of which are also
+/// converted. Maps and Iterables are copied to a new JavaScript object.
+/// Primitives and other transferable values are directly converted to their
+/// JavaScript type, and all other objects are proxied.
+dynamic jsifyAndAllowInterop(object) {
```
Comment (greglittlefield-wf) [inline lib/react_client/js_interop_helpers.dart]: We need to add test coverage for this function

File: test/js_interop_helpers_test/shared_tests.dart [hunk 2]
```diff
@@ -49,7 +57,7 @@ void sharedJsFunctionTests() {
 
     group('markChildValidated', () {
       test('is function that does not throw when called', () {
-        expect(() => markChildValidated(new EmptyObject()), returnsNormally);
+        expect(() => markChildValidated(newObject()), returnsNormally);
```
Comment (kealjones-wk) [inline test/js_interop_helpers_test/shared_tests.dart]: lets actually leave both tests in for now until its fully removed.
```suggestion
        expect(() => markChildValidated(new EmptyObject()), returnsNormally);
        expect(() => markChildValidated(newObject()), returnsNormally);
```

File: test/react_test_utils_test.dart [hunk 3]
```diff
@@ -220,7 +220,7 @@ void main() {
     });
 
     test('returns false argument is not an element', () {
-      expect(isElement(new EmptyObject()), isFalse);
+      expect(isElement(newObject()), isFalse);
```
Comment (kealjones-wk) [inline test/react_test_utils_test.dart]:
```suggestion
      expect(isElement(new EmptyObject()), isFalse);
      expect(isElement(newObject()), isFalse);
```

File: test/react_test_utils_test.dart [hunk 4]
```diff
@@ -3,10 +3,10 @@ library react_test_utils_test;
 
 import 'dart:html';
 
+import 'package:js/js_util.dart';
 import 'package:react/react.dart';
 import 'package:react/react_dom.dart' as react_dom;
 import 'package:react/react_client.dart';
-import 'package:react/react_client/js_interop_helpers.dart';
```
Comment (kealjones-wk) [inline test/react_test_utils_test.dart]:
```suggestion
import 'package:react/react_client/js_interop_helpers.dart';
```

File: test/util.dart [hunk 5]
```diff
@@ -2,7 +2,7 @@
 library react.test.util;
 
 import 'package:js/js.dart';
-import 'package:react/react_client/js_interop_helpers.dart';
+import 'package:js/js_util.dart';
```
Comment (kealjones-wk) [inline test/util.dart]: so package:js/js_util.dart actually just exports `dart:js_util` so id say using it is better than depending on an external library. but you could also just keep using js_interop_helpers.dart until its fully removed too. your call.
```suggestion
import 'dart:js_util';
```

File: test/react_client_test.dart [hunk 6]
```diff
@@ -34,7 +34,6 @@ main() {
           'children': testChildren,
         }),
       );
-      expect(unconvertJsProps(instance)['style'], new isInstanceOf<Map<String, dynamic>>());
```
Comment (kealjones-wk) [inline test/react_client_test.dart]: why was this removed?

File: test/js_interop_helpers_test/shared_tests.dart [hunk 7]
```diff
@@ -4,6 +4,8 @@ library js_function_test;
 import 'dart:html';
 
 import 'package:js/js.dart';
+import 'package:js/js_util.dart' show newObject;
```
Comment (kealjones-wk) [inline test/js_interop_helpers_test/shared_tests.dart]: so package:js/js_util.dart actually just exports `dart:js_util` so id say using it is better than depending on an external library. but you could also just keep using js_interop_helpers.dart until its fully removed too. your call.
```suggestion
import 'dart:js_util';
```

File: test/factory/common_factory_tests.dart [hunk 8]
```diff
@@ -1,12 +1,12 @@
 import 'dart:js';
 
+import 'package:js/js_util.dart';
```
Comment (kealjones-wk) [inline test/factory/common_factory_tests.dart]: so package:js/js_util.dart actually just exports `dart:js_util` so id say using it is better than depending on an external library. but you could also just keep using js_interop_helpers.dart until its fully removed too. your call.
```suggestion
import 'dart:js_util';
```

File: test/react_client_test.dart (hunk 6 above)
Comment (joebingham-wk) [reply test/react_client_test.dart]: Good catch - I didn't mean to. I was combining the branch that had the updated test cases with the master branch and this got cut out.

File: lib/react_client/js_interop_helpers.dart [hunk 9]
```diff
@@ -4,6 +4,7 @@
 library react_client.js_interop_helpers;
 
 import "package:js/js.dart";
+import "package:js/js_util.dart" as js_util;
```
Comment (kealjones-wk) [inline lib/react_client/js_interop_helpers.dart]:
```suggestion
import "dart:js_util" as js_util;
```

File: lib/react.dart [hunk 10]
```diff
@@ -328,11 +328,12 @@ abstract class Component {
   bool shouldComponentUpdate(Map nextProps, Map nextState) => true;
 
   /// > __DEPRECATED - DO NOT USE__
-  /// >
+  ///
```
Comment (kealjones-wk) [inline lib/react.dart]: Why were these removed?

Comment (joebingham-wk) [reply lib/react.dart]: No deliberate reason - just didn't think to add them on the empty lines when I pulled the comment back in. I updated it!

File: .gitignore [hunk 11]
```diff
@@ -6,3 +6,4 @@ pubspec.lock
 .pub/
 .packages
 .dart_tool
+/node_modules
```
[the remark this answers is not in the document]
Comment (joebingham-wk) [reply .gitignore]: Ah, I realized what happened. I had node modules from working on the 5.0.0-wip branch where they're already not being tracked, so when I started working on master it brought them over. I thought it was useful here, but I see why it's not. I'll take it out!

File: test/factory/common_factory_tests.dart [hunk 12]
```diff
@@ -82,7 +85,7 @@ void commonFactoryTests(ReactComponentFactoryProxy factory) {
   });
 }
 
-void domEventHandlerWrappingTests(ReactDomComponentFactoryProxy factory) {
+void domEventHandlerWrappingTests(ReactComponentFactoryProxy factory) {
```
[the remark this answers is not in the document]
Comment (joebingham-wk) [reply test/factory/common_factory_tests.dart]: I went back and forth on this line. I was pulling test code from a [commit Greg made](https://github.com/greglittlefield-wf/react-dart/commit/273b9152bff3455245e26f846d8fb07a915a1b09#diff-7a382a75d4df367a3f5df34d837d9227) a month ago (which was the source of the function in `js_interop_helpers.dart`) and came across this line. Since it was a month ago it seemed like a recent deliberate change in the type so I brought it over. Did I read the commit wrong or is still `ReactDomComponentFactoryProxy` preferred?

**Comment by kealjones-wk** [comment]
QA +1 
however, the title, and description are inaccurate due to recent need changes.
